android/src/main/java/com/zoontek/rnpermissions/RNPermissionsModule.java

```diff
@@ -145,10 +148,27 @@ private boolean permissionExists(final String permission) {
     }
   }
 
+  private boolean areNotificationsEnabled() {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        NotificationManager manager = (NotificationManager) getReactApplicationContext().getSystemService(Context.NOTIFICATION_SERVICE);
+        if (!manager.areNotificationsEnabled()) {
+          return false;
+        }
+        List<NotificationChannel> channels = manager.getNotificationChannels();
+        for (NotificationChannel channel : channels) {
+          if (channel.getImportance() == NotificationManager.IMPORTANCE_NONE) {
```

Comment:
What I am curious about is how this works when:
- you have multiple channels, and a user has only made one of them silent. Notifications may still be granted just not on one channel, isn't that still okay? it does not seem to justify a "false" here

This seems to start in the direction of something semantically different then "notifications are granted" and starts being "user has all notification channels enabled" which is really a different thing, since you could just create a new notification channel I think?

Is there a reference to the actual NotificationCompat code that shows the problem requiring this workaround on these API levels? And since NotificationCompat is actually in androidx.core (which is upgradable) I would expect to see a similar google issue tracker bug showing where the problem is discovered and fixed, and then likely it would just be an androidx.core minimum version bump with documentation in the changelog explaining why?

Reply:
You are right. Actually, I have chosen the implementation that if at least a notif channel is silenced, then the result of that method is false (as I assume it's rare to implement multiple channels on android rn).

If we want to be absolutely right we have to add some methods here to then be able to check for every channel if notifs are enabled or not. I don't think that it's a really useful case, is it?

Reply:
That's a good use case, but I don't think this is something this package should handle. It's the job of a "notifications-oriented" package imho (like https://notifee.app/)